theseus/utils/examples/bundle_adjustment/util.py

```diff
@@ -0,0 +1,63 @@
+import math
+from typing import Tuple
+
+import torch
+
+
+# ------------------------------------------------------------ #
+# --------------------------- LOSSES ------------------------- #
+# ------------------------------------------------------------ #
+def soft_loss_cauchy(
+    x: torch.Tensor, radius: torch.Tensor
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    ratio = (x + radius) / radius
+    val = torch.log(ratio) * radius
+    der = 1.0 / ratio
+    return val, der
+
+
+def soft_loss_huber_like(
+    x: torch.Tensor, radius: torch.Tensor
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    ratio = (x + radius) / radius
+    sq = torch.sqrt(ratio)
+    val = (sq - 1) * radius
+    der = 0.5 / sq
+    return val, der
+
+
+# ------------------------------------------------------------ #
+# ----------------------------- RNG -------------------------- #
+# ------------------------------------------------------------ #
+# returns a uniformly random point of the 2-sphere
+def random_s2(dtype: torch.dtype = torch.float64) -> torch.Tensor:
```

Comment:
Can we replace with the class version?

Reply:
cc: @maurimo

Reply:
certainly - this can be cleaned altogether, it's dangling code and not even used in the bundle adjustment IIRC!

Reply:
It looks like these are still used. I don't quite understand the difference between this sampling and the one @fantaosha  implemented in `.rand()`, so in the interest of time I propose to leave this as is for the time being.